### DISCUSSION TITLE
NixOS systemd service; recommended?

### DISCUSSION THREAD

**Discussion by ndom91** [ErikReider/SwayNotificationCenter#400] [Q&A]
Hey folks, thanks for this great notification center!

I've been launching `swaync` through Hyprland's `exec-once` on startup. But I recenlty noticed that a user systemd service is shipped with the `swaynotificationcenter` nixpkg.

Do you recommend using that service file on nixos? If so, how would you enable it declaratively? 

I'm relatively new to nixos, but setting `systemd.services.swaync.enable = true;` in `configuration.nix` doesn't seem to work, I assume it's got to be under `home.nix` or an import thereof to target the user systemd service, isntead of a (non-existent) system wide one? 

**EDIT**: Nevermind, it does seem to work like so:

```sh
systemd.user.units.swaync.enable = true;
```

**Comment by ndom91** [comment]
Ah okay, nevermind. After a few reboots (it seemed to work the first time :thinking:) It's now errorign with `cannot open display :0`

![image](https://github.com/ErikReider/SwayNotificationCenter/assets/7415984/6265ca38-0899-4189-a272-a45f1c79cca2)

I assume I have to add `WantedBy` or `After` for `hyprland.service`.. will report back

**EDIT**: Hmm it already ships with `After=graphical-session.target` :thinking: 

**EDIT 2**: I'm on systemd 254, so I'm assuming the `CONDITION_ENVIRONMENT` should be working too :thinking: :thinking: 

will do some more digging..

**Comment by ErikReider** [comment]
I'm sadly not a Nix user, so I won't really be of much help. I launch swaync through sways exec command, the same as Hyprlands `exec-once` command

**Comment by ndom91** [comment]
Hmm so after some more reboots (nothing changed) it seems to work most (all of?) the time. Maybe there was just some temporary race condition with something else :shrug:

Oh well, i'll close this for now then!

**Reply by lezuja** [reply]
Does it still work for you? I cannot see any systemd service in the current swaync nixpkg. A  manual systemd service ends up in the same errors as you showed before with unknown displays etc.
I ended up launching it through the window manager launch options.
